### DISCUSSION TITLE
install with different port for mysql

### DISCUSSION THREAD

**Discussion by ghost** [postalserver/postal#1839] [Help with using Postal]
how would i go about running postal initalize cmd while haveing the mysqlserver on an alternate port

**Comment by willpower232** [comment]
You should have been able to configure it in the config file before running `postal initialize`

**Comment by ghost** [comment]
I was just going over some things and I think it was my docker setup default security that was preventing access I assumed that the port must have been referred to in one of the setup file thinking the postal initialise CMD was expecting port 3306 my bad I'm in the process of writing a script to run on a server distro called nethserver that will ask some basic questions and automate the rest

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by ghost** [reply]
unfortunately i get an error

URI::InvalidURIError: the scheme mysql2 does not accept registry part: root:postal@127.0.0.1:3307:3306

in my script just before issuing postal initalize it has this directive


echo -e
echo  "Now I need to create the database and password for your postal instance"
echo -e
sleep 0.2s

echo -e
echo  "What port would you like to run mariadb on to avoid conflict?"
echo -e
read -r dbport
sleep 0.5s

config set postaldbserver service status enabled TCPPorts $dbport UDPPorts $dbport access red,green

signal-event firewall-adjust

echo -e
echo  "What would you like to call your database (ie postal)?"
echo -e
read -r db
sleep 0.5s

echo -e
echo  "Please enter a strong password?"
echo -e
read -r dbpw
sleep 0.5s

echo -e
echo "Thank you please wait while I create your database and credentials this may take a moment"
echo -e
sleep 0.2s

docker run -d \
   --name postal-mariadb \
   -p 127.0.0.1:$dbport:3306 \
   --restart always \
   -e MARIADB_DATABASE=$db \
   -e MARIADB_ROOT_PASSWORD=$dbpw \
   mariadb

wait

sed -i '26s/host: 127.0.0.1/host: 127.0.0.1:'"$dbport"'/g' /opt/postal/config/postal.yml;
sed -i '34s/host: 127.0.0.1/host: 127.0.0.1:'"$dbport"'/g' /opt/postal/config/postal.yml;

**Reply by willpower232** [reply]
~This sounds like you are using an unofficial script instead of following the official instructions so you might have a better time if you do that instead.~

https://docs.postalserver.io/

Edit: helps if I read the original post. If you're making this for yourself, it would probably be easier to write the configuration file manually and keep a copy of that. Alternatively you might have better luck editing the file with yq rather than sed so values can be set correctly.

**Comment by ghost** [comment]
Thanks sorry for that on closer look I think it's not postal related and more an issue with my implementation (or attempt at implementation) of mariadb on the plus side if I leave the port as default my script works fine

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
